src/python/pants/backend/project_info/filedeps.py

```diff
@@ -84,6 +85,8 @@ async def file_deps(
     if filedeps_subsystem.transitive:
         transitive_targets = await Get(TransitiveTargets, Addresses, addresses)
         targets = transitive_targets.closure
+    elif filedeps_subsystem.globs:
+        targets = await Get(UnexpandedTargets, Addresses, addresses)
```

Comment:
`./pants filedeps --transitive --globs` won't behave how I think it should. 

Without `--transitive`, we get this:

```
▶ ./pants filedeps --globs src/python/pants/util

src/python/pants/util/*.py
src/python/pants/util/BUILD
```

With `--transitive`, note that `*.py` no longer shows up:

```
▶ ./pants filedeps --globs --transitive src/python/pants/util

3rdparty/python/BUILD
3rdparty/python/requirements.txt
src/python/pants/BUILD
src/python/pants/__init__.py
src/python/pants/util/BUILD
src/python/pants/util/__init__.py
src/python/pants/util/collections.py
src/python/pants/util/contextutil.py
src/python/pants/util/dirutil.py
src/python/pants/util/enums.py
src/python/pants/util/eval.py
src/python/pants/util/filtering.py
src/python/pants/util/frozendict.py
src/python/pants/util/logging.py
src/python/pants/util/memo.py
src/python/pants/util/meta.py
src/python/pants/util/objects.py
src/python/pants/util/ordered_set.py
src/python/pants/util/osutil.py
src/python/pants/util/rwbuf.py
src/python/pants/util/socket.py
src/python/pants/util/strutil.py
```

Ideally, we would have a type of `UnexpandedTransitiveTargets` for this. I doubt it's worth that complexity, though.

Reply:
One thing that I think we should consider is removing the `--globs` flag, and/or moving it to another goal that dumps _any_ field from a target.

Reply:
There's a long-standing issue to "peek" at the target definition https://github.com/pantsbuild/pants/issues/4861. This might especially be helpful in the age of generated subtargets.

Reply:
Maybe... it feels like there is a possible symmetry with `filter`, where anything we can render we can filter on, and vice-versa?